### PR TITLE
Document how to host services at MOJ

### PR DESCRIPTION
**Pull request by SteveMarshall**
It's long been the case that this was just something we tried to socialise, but that hasn't been entirely successful. This tries to make explicit some of the things that are mostly in my head, and documents the reasoning behind them.

### CODE REVIEW

**Review by digitalronin — CHANGES_REQUESTED** [review]
The file needs to be named `hosting.html.md.erb` so that the published version has the docpath `/documentation/standards/hosting.html`

File: source/documentation/standards/hosting.md.erb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+title: How to host services
+last_reviewed_on: 2020-09-01
+review_in: 3 months
+---
+
+# <%= current_page.data.title %>
+
+At MOJ you should use the [Cloud
```
Comment (jakemulley) [inline source/documentation/standards/hosting.md.erb:9]: If we want to set the precedence of using the Cloud Platform as the default (and not using a separate AWS account, for example), would it be better to change "should" to "must"?

File: source/documentation/standards/hosting.md.erb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+by default for scalable computing, storage and deployment services, but
+we may recommend some services are hosted in [Microsoft
+Azure](https://azure.microsoft.com/), based on the needs of the service
+and its relationship to other services already in Azure.
+
+If your service is currently hosted outside of AWS or Azure, get in
+touch with [the Application Modernisation team](#TODO) to plan for its
+future hosting.
+
+## Paying for hosting
```
Comment (SteveMarshall) [inline source/documentation/standards/hosting.md.erb:54]: This doesn't fit the tense of the other heading; it should be `We pay for hosting centrally by default`

File: source/documentation/standards/hosting.md.erb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  environments
+
+We currently use [Amazon Web Services (AWS)](https://aws.amazon.com/)
+by default for scalable computing, storage and deployment services, but
+we may recommend some services are hosted in [Microsoft
+Azure](https://azure.microsoft.com/), based on the needs of the service
+and its relationship to other services already in Azure.
+
+If your service is currently hosted outside of AWS or Azure, get in
+touch with [the Application Modernisation team](#TODO) to plan for its
```
Comment (SteveMarshall) [inline source/documentation/standards/hosting.md.erb:51]: I need an email address for app modernisation…

File: source/documentation/standards/hosting.md.erb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+review_in: 3 months
+---
+
+# <%= current_page.data.title %>
+
+At MOJ you should use the [Cloud
+Platform](https://user-guide.cloud-platform.service.justice.gov.uk/) to
+host your service unless you're advised otherwise by a member of [the
+hosting team](https://peoplefinder.service.gov.uk/teams/hosting). Your
+team can get started on the Cloud Platform without letting us know;
```
Comment (jakemulley) [inline source/documentation/standards/hosting.md.erb:13]: This should probably be more explicit to who "us" is - Hosting team, or Cloud Platform team?

**Review by digitalronin — CHANGES_REQUESTED** [review]
the content looks great. As well as the filename change, you also need to add a link on the index page `source/documentation/index.html.md.erb`

File: source/documentation/standards/hosting.md.erb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+touch with the team if you need help.
+
+If you think that your service might require other hosting, [get in
+touch with the hosting
+team](mailto:hosting-leads@digital.justice.gov.uk) and we'll help
+identify which of our other hosting options is best for your service,
+or set up extra accounts if we've already identified the right
+environment for you:
+
+- AWS Modernisation Platform (currently in alpha)
```
Comment (SteveMarshall) [inline source/documentation/standards/hosting.md.erb:25]: “alpha” might be a bit optimistic… But maybe we can use this as a call to action? @jakemulley waddya think?

**Review by SteveMarshall — COMMENTED** [review]
I apparently can't “request changes” to my own PR, which is a shame, because this needs some changes.

**Review by jakemulley — CHANGES_REQUESTED** [review]

File: source/documentation/standards/hosting.md.erb (hunk 5 above)
Comment (jakemulley) [reply source/documentation/standards/hosting.md.erb]: I think alpha is too optimistic & agree with the CTA. Would be good to direct people to talk to us.

Maybe this list should be ordered on basis of preference if it's not already. I'm unsure if self-managed is something we'd want to push?

**Review by AntonyBishop — COMMENTED** [review]
Makes sense to me.

**Review by jakemulley — APPROVED** [review]

**Comment by davidread** [comment]
This is super 👍 

As a bonus, I'm wondering about a few words that sells Cloud Platform as a great choice - i.e. some "carrot" to balance with the stick "you must use it". There is the section about public cloud, which is great, but misses some of the extra value of Cloud Platform.

I'm thinking something along the lines:

> Cloud Platform is a modern hosting platform for digital services:
> * It makes it quick and easy to 'just deploy an app' and for free you get easy scalability, zero downtime deploys, security scanning, monitoring, logging etc, all setup for you by default
> * When you need the full power of AWS infrastructure or Kubernetes features, it's all available
> * Config is all self-service, using Infrastructure as Code - GitOps best practice - with managed CI/CD provided
> * The Cloud Platform team manage the hosting and provide support

This might be better located in the CP docs. Perhaps there's some better text already written somewhere and I've just not found it yet? @AntonyBishop @digitalronin

**Review by davidread — APPROVED** [review]

File: source/documentation/standards/hosting.md.erb (hunk 5 above)
Comment (SteveMarshall) [reply source/documentation/standards/hosting.md.erb]: I've added a bit more context on Modernisation Platform, and re-ordered the list a little. I don't want to _push_ self-managed, you're right, but I think it should at least be listed…

**Comment by SteveMarshall** [comment]
> As a bonus, I'm wondering about a few words that sells Cloud Platform as a great choice - i.e. some "carrot" to balance with the stick "you must use it". There is the section about public cloud, which is great, but misses some of the extra value of Cloud Platform.

I've pretty much lifted what you wrote 👍; @AntonyBishop if you have changes you'd like, let me know.

File: source/documentation/standards/hosting.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+The Cloud Platform is a modern hosting platform for digital services:
+
+- It makes it quick and easy to 'just deploy an app' and you get easy
+  scalability, zero downtime deploys, security scanning, monitoring,
+  logging, and more, all setup for you by default
+- When you need the full power of AWS or Kubernetes, it's all available
+- Configuration is all self-service, using Infrastructure as Code with
+  managed CI/CD and GitOps processes
+- The Cloud Platform team manage the hosting and provide support
```
Comment (AntonyBishop) [inline source/documentation/standards/hosting.html.md.erb:26]: add "for the platform"

File: source/documentation/standards/hosting.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+host your service unless you're advised otherwise by a member of [the
+hosting team](https://peoplefinder.service.gov.uk/teams/hosting). Your
+team can get started on the Cloud Platform without letting anyone know;
+just follow [the Cloud Platform documentation
+](https://user-guide.cloud-platform.service.justice.gov.uk/) and get in
+touch with the Cloud Platform team if you need help.
+
+The Cloud Platform is a modern hosting platform for digital services:
+
+- It makes it quick and easy to 'just deploy an app' and you get easy
```
Comment (AntonyBishop) [inline source/documentation/standards/hosting.html.md.erb:20]: Can we split this into two paras to emphasise deploying and operating a service. Maybe something like:

- It makes it quick and easy to 'just deploy an app'
- Provides team with the features to operate their service, such as scalability, zero downtime deploys, security scanning, monitoring, logging, and more, all setup for you with sensible defaults

**Review by AntonyBishop — APPROVED** [review]

**Comment by davidread** [comment]
👍 👍 👍

**Comment by jakemulley** [comment]
:shipit:

**Review by digitalronin — APPROVED** [review]

**Pull request merged**